Book-A-Desk.Api/BookingsHttpHandler.fs

```diff
@@ -14,20 +15,31 @@ open Book_A_Desk.Domain.Office.Domain
 open Book_A_Desk.Domain.Reservation
 open Book_A_Desk.Domain.Reservation.Commands
 open Book_A_Desk.Domain.Reservation.Domain
+
 open Book_A_Desk.Infrastructure.DynamoDbEventStore
 
 type BookingsHttpHandler =
     {
         HandlePostWith: Models.Booking -> HttpHandler
     }
 
+type EmailDetails =
+    {
+        toAddress : string
+        fromAddress : string
+        body : string
+    }
+
+type MessageDetails =
+    | Email of EmailDetails
+
 module BookingsHttpHandler =
-    let initialize (provideEventStore : IAmazonDynamoDB -> DynamoDbEventStore) reservationCommandsFactory =
+    let initialize (provideEventStore : IAmazonDynamoDB -> DynamoDbEventStore) reservationCommandsFactory sendEmailNotification =
         let handlePostWith booking = fun next (context : HttpContext) ->
             task {
                 let cmd =
                     {
-                        BookADesk.OfficeId = Guid.Parse(booking.Office.Id) |> OfficeId // Consider TryParse and return 400 if not valid
+                        BookADesk.OfficeId = booking.Office.Id
```

Comment:
As the API specs says that the office id is a string I think this is not correct anymore here and the previous implementation was correct.